carmm/analyse/traj_to_gif.py

```diff
@@ -0,0 +1,106 @@
+from ase.io import read, write
+from ase.visualize import view
+import os
+
+def traj_to_gif(filename, frames_per_second=30, pause_time=0.5, atom_subs=None, keep_temp_files=True, test=False):
+
+    '''
+    A function which takes a .traj file, visualises it in povray with your desired settings and outputs a .gif file.
+    When the function gives you the ase viewer, rotate to your desired view, go to Tools -> Render Scene, select the "Render all frames" option and deselect "Show output window".
+    MAKE SURE THAT THE OUTPUT BASENAME IS THE SAME AS THE INITIAL FILENAME (e.g. for 'atoms.traj' the output basename is 'atoms')
+    The rest of the settings can be changed at your discretion.
+    Then press "Render" and wait for it to finish before closing the window and pressing enter to continue.
+    The ImageMagick linux suite is required for this function.
+    :param filename: (str) Full name/directory of the .traj file to convert
+    :param frames_per_second: (float) Speed of switching images (Default is 30 fps)
+    :param pause_time: (float) Time to pause on the first and last images (Default is 0.5 seconds)
+    :param atom_subs: (list of lists of strings) Pairs of atomic symbols with the first being changed to the second in all
+    images for clearer visualisation
+    :param keep_temp_files: (boolean) If set to False, will delete the created povray .traj and .png files after use
+    :param test: (boolean) Do not change! A quick fix to stop the unittest from getting stuck waiting for a user input
+    :return: A .gif file of the .traj file, visualised in povray
+    '''
+
+    #Retrieve the file name and file extension from (potentially) a full directory path
+    if filename == None:
+        filename = 'atoms.traj'
+    file = filename.split('/')[-1]
+    file, ext = file.split('.')
+
+    atoms = read(filename+'@:')
+    steps = len(atoms)
+
+    if not test:
+        povray_render(atoms, steps, file, ext, atom_subs)
+
+    gifmaker(steps, file, ext, frames_per_second, pause_time, keep_temp_files, test)
+
+    print("Happy cooking!")
+
+    # For testing purposes
+    return file, ext, steps
+
+
+def atom_sub(atoms, atom_subs, steps, file, ext):
```

Comment:
I would define atom_subs as a list of lists to allow multiple substitutions if required.

Reply:
The code already looks for a list of lists of strings as the atom_subs input.

Will change the example traj_to_gif function (the top comment in examples/analyse_traj_to_gif.py) to show a case where there are multiple atom substitutions to show this more clearly.